tests/test_twitter_ads_bookmark.py

```diff
@@ -1,179 +1,179 @@
+                        second_bookmark_value)
+
+
+                    simulated_bookmark_value = self.convert_state_to_utc(new_states['bookmarks'][stream][self.account_id])
+
+                    # Verify the first sync sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_value)
+
+                    # Verify the second sync sets a bookmark of the expected form
+                    self.assertIsNotNone(second_bookmark_value)
+
+                    # Verify the second sync bookmark is Equal to the first sync bookmark
+                    # assumes no changes to data during test
+                    self.assertEqual(second_bookmark_value, first_bookmark_value)
+
+                    for record in first_sync_messages:
+
+                        # Verify the first sync bookmark value is the max replication key value for a given stream
+                        replication_key_value = self.convert_state_to_utc(record.get(replication_key))
+
+                        self.assertLessEqual(
+                            replication_key_value, first_bookmark_value_utc,
+                            msg="First sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
+                        )
+
+                    for record in second_sync_messages:
+                        # Verify the second sync replication key value is Greater or Equal to the first sync bookmark
+                        replication_key_value = self.convert_state_to_utc(record.get(replication_key))
+
+                        self.assertGreaterEqual(replication_key_value, simulated_bookmark_value,
+                                                msg="Second sync records do not repect the previous bookmark.")
+
+                        # Verify the second sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(
+                            replication_key_value, second_bookmark_value_utc,
+                            msg="Second sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
+                        )
+
+                elif expected_replication_method == self.FULL_TABLE:
+
+                    # Verify the syncs do not set a bookmark for full table streams
+                    self.assertIsNone(first_bookmark_value)
+                    self.assertIsNone(second_bookmark_value)
+
+                    # Verify the number of records in the second sync is the same as the first
+
+                    # `targeting_criteria` is child streams of parent stream `line_items` and `line_items` is an incremental streams
+                    # Child stream also behave like incremental streams but does not save it's own state. So, it don't
+                    # have same no of record on second sync and first sync.
+                    if stream in ['targeting_criteria']:
+                        continue
```

Comment:
The `continue` here is skipping the final check that at least 1 record was replicated on sync 2. Can this be updated to skip only the assertion on line 169?

Reply:
Updated test case to skip the assertion on line 169.